python/setup_project/storage_map_widget.py

```diff
@@ -453,15 +467,83 @@ def _create_new_storage(self):
             # will switch away from "create" back to "choose" or a storage.
             self.guess_storage()
 
-    def _on_path_changed(self, path, edit_lookup):
+    def _on_path_changed(self, path, platform):
         """
-        Keep track of any path edits as they happen. Kepe the user informed if
+        Keep track of any path edits as they happen. Keep the user informed if
         there are any concerns about the entered text.
         """
 
-        # store the edited path in the appropriate path lookup
+        # does the path only contain slashes?
+        only_slashes = path.replace("/", "") == "" or \
+            path.replace("\\", "") == ""
+
+        # does it end in a slash?
+        trailing_slash = path.endswith("/") or path.endswith("\\")
+
+        # the name of the storage being edited
         storage_name = str(self.ui.storage_select_combo.currentText())
-        edit_lookup[storage_name] = path
+
+        # a temp SG path object used for sanitization
+        sg_path = ShotgunPath()
+
+        # store the edited path in the appropriate path lookup. sanitize first
+        # by running it through the ShotgunPath object. since sanitize removes
+        # the trailing slash, add it back in if the user typed it.
+        # if the sanitized path differs, update the edit.
+        if platform.startswith("linux"):
+            if only_slashes:
+                # SG path code doesn't like only slashes in a path
+                self._linux_path_edit[storage_name] = path
+            elif path:
+                sg_path.linux = path  # sanitize
+                sanitized_path = sg_path.linux
+                if trailing_slash:
+                    # add the trailing slash back in
+                    sanitized_path = "%s/" % (sanitized_path,)
```

Comment:
why do we need this? would it be possible to detect if a path changed based on the sanitized paths rather than their raw string value? ShotgunPaths support the equality operator.

Reply:
because of the sanitize, any trailing slashes get removed. since this method is called whenever the path is changed by the user, it's impossible to manually type a trailing slash followed by some more text. So we want to sanitize it on the fly and make sure everything the user typed is valid, but we store it raw for cases like this where the path is valid and sanitized, but maybe not finished.